driver-core/src/main/com/mongodb/MongoCredential.java

```diff
@@ -287,14 +314,13 @@ public static MongoCredential createGSSAPICredential(final String userName) {
     /**
      * Creates a MongoCredential instance for the MONGODB-AWS mechanism.
      *
-     * @param userName the user name
-     * @param password the user password
+     * @param userName the user name, which may be null.  This maps to the AWS accessKeyId
+     * @param password the user password, which may be null if the userName is also null.  This maps to the AWS secretAccessKey.
```

Comment:
I'd rather not make this change now.  The way AwsAuthenticator is written, it could actually work the way it is, and I don't want to risk introducing a breaking change.  But I'll check the spec and open a separate ticket if we're missing validation.